src/apps/file/api/file.py

```diff
@@ -65,18 +73,28 @@ async def download(
 
 async def answer_upload(
     applet_id: uuid.UUID,
+    file_id: str,
     file: UploadFile = File(...),
     user: User = Depends(get_current_user),
     session: AsyncSession = Depends(get_session),
 ):
+    if not await UserAppletAccessCRUD(session).get_by_roles(
+        user.id,
+        applet_id,
+        [Role.OWNER, Role.MANAGER, Role.REVIEWER],
+    ):
+        raise AnswerViewAccessDenied()
+
     cdn_client = await select_storage(applet_id, session)
-    key = cdn_client.generate_key(hash(user.id), file.filename)
+    unique = f"{user.id}/{applet_id}"
+    cleaned_file_id = file_id.strip()
```

Comment:
file_id is not required here. If it exists, you should use it, if not - use filename with some unique to avoid rewritting